app/models/manageiq/providers/redfish/inventory/parser/physical_infra_manager.rb

```diff
@@ -99,5 +114,51 @@ def get_storage_sum(server)
     def get_storage_capacity(storage)
       storage.Drives.reduce(0) { |acc, d| acc + (d.CapacityBytes || 0) }
     end
+
+    def physical_racks
+      collector.physical_racks.each do |r|
+        persister.physical_racks.build(
+          :ems_ref => r["@odata.id"],
+          :name    => r.Id
+        )
+      end
+    end
+
+    def physical_chassis
+      collector.physical_chassis.each do |c|
+        chassis = persister.physical_chassis.build(
+          :ems_ref      => c["@odata.id"],
+          :health_state => c.Status.Health,
+          :name         => c.Id,
+          :type         => "ManageIQ::Providers::Redfish::PhysicalInfraManager::PhysicalChassis",
```

Comment:
Try leaving the type out and see if it gets the right STI class, we should be automatically building this now.